go/stellar/stellar.go

```diff
@@ -1590,6 +1590,21 @@ func SetAccountAsPrimary(m libkb.MetaContext, walletState *WalletState, accountI
 	return walletState.UpdateAccountEntriesWithBundle(m, "set account as primary", &nextBundle)
 }
 
+func DeleteAccountIfEmpty(mctx libkb.MetaContext, remoter remote.Remoter, accountID stellar1.AccountID) error {
+	funded, err := isAccountFunded(mctx.Ctx(), remoter, accountID)
```

Comment:
This uses caching when it shoudn't. `remoter` is a `WalletState`. This is where wallet state being sneaked into remoter is confusing. (I could have it wrong)

Reply:
hah. so, i noticed that, and there's a wallet state refresh right before this is called. but yes, i agree. i'd rather just not cache. do you think it's better to pass in the wallet state (instead of remoter) and refresh it in here? or is there something cleaner?

Reply:
Ah nice.

1. I think ideally `WalletState` wouldn't implement `Remoter`. But that's probably too much trouble. 
2. `Server` has `.remoter` which is actually WalletState and `.walletState` which is the same (weird). Could make `.remoter` an actual remoter and change existing uses of `.remoter` over to `.walletState`.
3. Or add `.remoterActual` and pass that into DeleteAccountIfEmtpy.
4. Or add a boolean to isAccountFunded to punch through caches.

Whatever you want to do including leaving at as is if the refresh works is cool with me. But if you leave just the refresh how about moving the Refresh call down into `DeleteAccountIfEmpty`? So that a potential future caller of `DeleteAccountIfEmpty` doesn't get a cache surprise 💸 .

Reply:
pretty sure this is fixed in a nice way now.